### PR TITLE
python312Packages.botorch: disable failing test on aarch64-darwin

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

Occurs only on `aarch64-darwin`.

```
E       AssertionError: Tensor-likes are not close!
E
E       Mismatched elements: 1 / 2 (50.0%)
E       Greatest absolute difference: 5.155801773071289e-06 at index (0, 0) (up to 1e-06 allowed)
E       Greatest relative difference: 1.655621781537775e-05 at index (0, 0) (up to 1e-05 allowed)
```

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352613`

---
### `x86_64-linux`

---
### `aarch64-linux`

---
### `x86_64-darwin`

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.ax-platform</li>
    <li>python311Packages.ax-platform.dist</li>
    <li>python312Packages.ax-platform</li>
    <li>python312Packages.ax-platform.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>python311Packages.botorch</li>
    <li>python311Packages.botorch.dist</li>
    <li>python311Packages.optuna</li>
    <li>python311Packages.optuna.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pyannote-pipeline</li>
    <li>python311Packages.pyannote-pipeline.dist</li>
    <li>python312Packages.botorch</li>
    <li>python312Packages.botorch.dist</li>
    <li>python312Packages.optuna</li>
    <li>python312Packages.optuna.dist</li>
    <li>python312Packages.pyannote-pipeline</li>
    <li>python312Packages.pyannote-pipeline.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352613`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.ax-platform</li>
    <li>python311Packages.ax-platform.dist</li>
    <li>python312Packages.ax-platform</li>
    <li>python312Packages.ax-platform.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.ax-platform</li>
    <li>python311Packages.ax-platform.dist</li>
    <li>python312Packages.ax-platform</li>
    <li>python312Packages.ax-platform.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.ax-platform</li>
    <li>python311Packages.ax-platform.dist</li>
    <li>python312Packages.ax-platform</li>
    <li>python312Packages.ax-platform.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>python311Packages.ax-platform</li>
    <li>python311Packages.ax-platform.dist</li>
    <li>python311Packages.botorch</li>
    <li>python311Packages.botorch.dist</li>
    <li>python311Packages.optuna</li>
    <li>python311Packages.optuna.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pyannote-pipeline</li>
    <li>python311Packages.pyannote-pipeline.dist</li>
    <li>python312Packages.ax-platform</li>
    <li>python312Packages.ax-platform.dist</li>
    <li>python312Packages.botorch</li>
    <li>python312Packages.botorch.dist</li>
    <li>python312Packages.optuna</li>
    <li>python312Packages.optuna.dist</li>
    <li>python312Packages.pyannote-pipeline</li>
    <li>python312Packages.pyannote-pipeline.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]

File: pkgs/development/python-modules/ax-platform/default.nix [hunk 1]
```diff
@@ -66,6 +68,10 @@ buildPythonPackage rec {
     "--ignore=ax/service/tests/test_scheduler.py"
     "--ignore=ax/service/tests/test_with_db_settings_base.py"
     "--ignore=ax/storage"
+    # E   sqlalchemy.exc.ArgumentError: Type annotation for "SQAMetric.scalarized_objective_children_metrics"
+    # can't be correctly interpreted for Annotated Declarative Table form.
+    # (Background on this error at: https://sqlalche.me/e/20/zlpr)
+    "--ignore=ax/core/tests/test_experiment.py"
```
Comment (veprbl) [inline pkgs/development/python-modules/ax-platform/default.nix:74]: This means that sqlalchemy 2+ is not supported, see #264724

Comment (GaetanLepage) [reply pkgs/development/python-modules/ax-platform/default.nix]: Thanks ! I wasn't sure.
I have removed the `sqlalchemy` dependency.

**Comment by veprbl** [comment]
@grahamcofborg build python312Packages.ax

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352613`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.ax-platform</li>
    <li>python311Packages.ax-platform.dist</li>
    <li>python312Packages.ax-platform</li>
    <li>python312Packages.ax-platform.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.ax-platform</li>
    <li>python311Packages.ax-platform.dist</li>
    <li>python312Packages.ax-platform</li>
    <li>python312Packages.ax-platform.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.ax-platform</li>
    <li>python311Packages.ax-platform.dist</li>
    <li>python312Packages.ax-platform</li>
    <li>python312Packages.ax-platform.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>python311Packages.ax-platform</li>
    <li>python311Packages.ax-platform.dist</li>
    <li>python311Packages.botorch</li>
    <li>python311Packages.botorch.dist</li>
    <li>python311Packages.optuna</li>
    <li>python311Packages.optuna.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pyannote-pipeline</li>
    <li>python311Packages.pyannote-pipeline.dist</li>
    <li>python312Packages.ax-platform</li>
    <li>python312Packages.ax-platform.dist</li>
    <li>python312Packages.botorch</li>
    <li>python312Packages.botorch.dist</li>
    <li>python312Packages.optuna</li>
    <li>python312Packages.optuna.dist</li>
    <li>python312Packages.pyannote-pipeline</li>
    <li>python312Packages.pyannote-pipeline.dist</li>
  </ul>
</details>

**Pull request merged**